### DISCUSSION TITLE
Help on LN part of LNP/BP Core Library

### DISCUSSION THREAD

**Discussion by UkolovaOlga** [LNP-BP/devcalls#32] [Organisational]
We need to implement the BOLTs as extensions. These extensions can be replaced later with other protocols. The structure of these extensions is already there. Now it's a matter of feeling in the details. If anyone wanna go down this rabbit hole with @rajarshimaitra , do reach out/leave your comments here. 

This is quite a big chunk of work and more help we can get its better. Raj will brief you over the basic structures that are already in place and what things still need to be done. Which mostly involves reading out the BOLTs and implementing them in rust.

Feel free to ask questions and/or raise a hand to help.

**Comment by rajarshimaitra** [comment]
Thanks @UkolovaOlga . Great feature by github. 

**Reference** 

this is where the Bolt3 extension lives. Which is an extension that creates the commitment transaction of a channel as per [BOLT3](https://github.com/lightningnetwork/lightning-rfc/blob/master/03-transactions.md).
https://github.com/LNP-BP/rust-lnpbp/blob/master/src/lnp/application/payment/constructors/bolt3.rs


I have opened a draft PR for the Htlc part, which will be responsible for adding the HTLC transactions. Going through the discussion might be helpful as I have pointed a few required design decisions for which I am looking for suggestions.
https://github.com/LNP-BP/rust-lnpbp/pull/163

And do check out this dev call presentation from few weeks back. to get an overall idea of what lnp-node is trying to achieve.  
https://www.youtube.com/watch?v=TgmyO0ecVNI

Contact me at telegram anytime if you are interested in implementing these extensions.

**Reply by UkolovaOlga** [reply]
This place is great to start any sort of discussion or research on the topic.
If you want to contribute and learn more details, feel free to jump in here or join our IRC channel (#lnp-np at Freenode).
